### PR TITLE
Improved sample gen

### PR DESCRIPTION
**Pull request by michaelbausor**
Added while loop access for unpaged list callables
Added futureCall usage to other callables
Other tweaks in main.snip

### CODE REVIEW

**Comment by michaelbausor** [comment]
@garrettjonesgoogle Re futureCall - this PR uses futureCall for callables returning a single object and for paged callables, but not for unpaged list callables (which use the while loop)
Callables with no return type (e.g. deleteTopic) use a ListenableFuture<Void>

File: src/main/java/com/google/api/codegen/java/JavaContextCommon.java [hunk 1]
```diff
@@ -202,6 +202,12 @@ public static Variable s_newVariable(
 
     public abstract String getReturnType();
 
+    public String getGenericAwareReturnType() {
+      String returnType = getReturnType();
+      if (returnType == null || returnType.isEmpty()) return "Void";
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/java/JavaContextCommon.java]: always use brackets for if statements

**Comment by michaelbausor** [comment]
PTAL

**Comment by garrettjonesgoogle** [comment]
LGTM

**Pull request merged**